hopp/tools/design/wind/turbine_library_interface_tools.py

```diff
@@ -0,0 +1,242 @@
+    
+    if has_cp_curve:
+        cp_curve = np.array(turbine_specs["power_curve"]["cp"].to_list())
+        cp_curve = np.nan_to_num(cp_curve)
+        cp_curve = np.where(cp_curve<0,0,cp_curve).tolist()
+
+    if has_power_curve and has_cp_curve:
+        power_curve_kw = np.array(turbine_specs["power_curve"]["power_kw"].to_list())
+        power_curve_kw = np.nan_to_num(power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw<0,0,power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw>turbine_specs["rated_power"],turbine_specs["rated_power"],power_curve_kw).tolist()
+        
+        cp_curve = np.array(turbine_specs["power_curve"]["cp"].to_list())
+        cp_curve = np.nan_to_num(cp_curve)
+        cp_curve = np.where(cp_curve<0,0,cp_curve).tolist()
+        
+    if has_cp_curve and not has_power_curve:
+        cp_curve = np.array(turbine_specs["power_curve"]["cp"].to_list())
+        cp_curve = np.nan_to_num(cp_curve)
+        cp_curve = np.where(cp_curve<0,0,cp_curve).tolist()
+        power_curve_kw = curve_tools.calculate_power_from_cp(wind_speeds,cp_curve,turbine_specs["rotor_diameter"],turbine_specs["rated_power"])
+    
+    if has_power_curve and not has_cp_curve:
+        power_curve_kw = np.array(turbine_specs["power_curve"]["power_kw"].to_list())
+        power_curve_kw = np.nan_to_num(power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw<0,0,power_curve_kw)
+        power_curve_kw = np.where(power_curve_kw>turbine_specs["rated_power"],turbine_specs["rated_power"],power_curve_kw).tolist()
+        
+        cp_curve = curve_tools.calculate_cp_from_power(wind_speeds,power_curve_kw)
+        
+    if has_ct_curve:
+        ct = turbine_specs["power_curve"]["ct"].to_list()
+    else:
+        ct = curve_tools.estimate_thrust_coefficient(wind_speeds,cp_curve)
+    
+    _, cp_curve = curve_tools.pad_power_curve(wind_speeds,cp_curve)
+    _, ct = curve_tools.pad_power_curve(wind_speeds,ct)
+    wind_speeds, power_curve_kw = curve_tools.pad_power_curve(wind_speeds,power_curve_kw)
+    
+    if model_name == "floris":
+        power_thrust_table = {
+            "wind_speed":wind_speeds,
+            "power":power_curve_kw,
+            "thrust_coefficient":ct,
+            }
+        return power_thrust_table
+    
+    # if model_name is "pysam"
+    power_thrust_table = {
+        "wind_turbine_max_cp": max(cp_curve),
+        "wind_turbine_ct_curve":ct,
+        "wind_turbine_powercurve_windspeeds":wind_speeds,
+        "wind_turbine_powercurve_powerout":power_curve_kw,
+        }
+    return power_thrust_table
+
+
+def check_hub_height(turbine_specs, wind_plant):
+    """Check the hub-height from the turbine-library specs against the other possible hub-height entries. 
+    If multiple hub-height options are available from the turbine_specs, it will choose one based on user-input information.
```

Comment:
Could you make this sentence more clear? Just by reading this, I'm not sure what user-inputted info controls this. I'm thinking "it" is "this method" but could you also clarify that?